core/state_transition.go

```diff
@@ -349,11 +358,23 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	if contractCreation {
 		ret, _, st.gas, vmerr = st.evm.Create(sender, st.data, st.gas, st.value)
 	} else {
-		// Increment the nonce for the next transaction
-		st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
+		// dont increment nonce for l1 messages
+		if !st.msg.IsL1MessageTx() {
+			// Increment the nonce for the next transaction
+			st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
```

Comment:
one weird detail here to note. we don't increment the nonce during a regular call , but do increment it if the tx creates a contract (we need to increment nonces in create bc of the address calculations). do we want that?

Reply:
Maybe we can increment the nonce in both cases but just skip the nonce check for `L1MessageTx`? @zimpha

Reply:
will reopen on next PR